pkg/tfshim/sdk-v2/internal/tf/plans/objchange/objchange.go

```diff
@@ -75,10 +75,7 @@ func PlannedDataResourceObject(schema *configschema.Block, config cty.Value) cty
 }
 
 func proposedNew(schema *configschema.Block, prior, config cty.Value) cty.Value {
-	if !config.IsKnown() {
-		return config
-	}
-	if config.IsNull() {
+	if config.IsNull() || !config.IsKnown() {
```

Comment:
This is the version in OpenTOFU. It is very suspect to deviate in this algo.

Reply:
Good catch, I don't believe this change was left intentionally - this was the initial attempt at fixing the issue in 2060 before we caught that upstream had updates.

Reply:
Does this fix anything on the PF side? How did you catch this?